### PR TITLE
[ONE]Implement Against All Odds

### PR DESCRIPTION
**Pull request by AhmadYProjects**

### CODE REVIEW

**Comment by AhmadYProjects** [comment]
[[against all odds]]

File: Mage.Sets/src/mage/cards/a/AgainstAllOdds.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import mage.target.common.TargetControlledPermanent;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author AhmadYProjects
+ */
+public final class AgainstAllOdds extends CardImpl {
+    private static final FilterControlledPermanent filter = new FilterControlledPermanent("creature card with mana value 3 or less");
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/a/AgainstAllOdds.java:25]: this filter shouldn't be mana value 3 or less

File: Mage.Sets/src/mage/cards/a/AgainstAllOdds.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                CardType.CREATURE.getPredicate(),
+                CardType.ARTIFACT.getPredicate()
+        ));
+
+    }
+    public AgainstAllOdds(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{3}{W}");
+        
+
+        // Choose one or both --
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/a/AgainstAllOdds.java:45]: don't forget to set the min and max modes

**Review by theelk801 — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/a/AgainstAllOdds.java (hunk 1 above)
Comment (AhmadYProjects) [reply Mage.Sets/src/mage/cards/a/AgainstAllOdds.java]: fixed

File: Mage.Sets/src/mage/cards/a/AgainstAllOdds.java (hunk 2 above)
Comment (AhmadYProjects) [reply Mage.Sets/src/mage/cards/a/AgainstAllOdds.java:45]: fixed

File: Mage.Sets/src/mage/cards/a/AgainstAllOdds.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+public final class AgainstAllOdds extends CardImpl {
+    private static final FilterControlledPermanent filter = new FilterControlledPermanent("artifact or creature card you control");
+    private static final FilterCard filter2 = new FilterCard("artifact or creature card with mana value 3 or less from your graveyard");
+
+    static{
+        filter.add(Predicates.or(
+                CardType.CREATURE.getPredicate(),
+                CardType.ARTIFACT.getPredicate()
+        ));
+        filter.add(new ManaValuePredicate(ComparisonType.FEWER_THAN, 4));
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/a/AgainstAllOdds.java:33]: this filter shouldn't have this predicate

**Review by theelk801 — CHANGES_REQUESTED** [review]

**Pull request merged**